test/membrane_http_adaptive_stream/integration_test/sink_bin_integration_test.exs

```diff
@@ -0,0 +1,189 @@
+          |> via_in(Pad.ref(:input, track_name),
+            options: [encoding: encoding, track_name: track_name]
+          )
+          |> to(:sink_bin)
+        end)
+
+      {{:ok, spec: %ParentSpec{children: children, links: links}}, %{}}
+    end
+  end
+
+  describe "Reference HLS content creation should only take place locally." do
+    test "check if fixture creation is unset" do
+      assert not @create_fixtures
+    end
+  end
+
+  describe "Testing HLS content creation for single media track." do
+    test "single video track" do
+      test_pipeline(
+        @single_video_track_source,
+        @single_video_track_ref_path,
+        @single_video_track_test_path
+      )
+    end
+  end
+
+  describe "Testing HLS content creation for single audio and single video tracks." do
+    test "audio and video tracks" do
+      test_pipeline(
+        @audio_video_tracks_sources,
+        @audio_video_tracks_ref_path,
+        @audio_video_tracks_test_path
+      )
+    end
+  end
+
+  describe "Testing HLS content creation for single audio and multiple video of various quality tracks." do
+    test "audio and multiple video tracks" do
+      test_pipeline(
+        @audio_multiple_video_tracks_sources,
+        @audio_multiple_video_tracks_ref_path,
+        @audio_multiple_video_tracks_test_path
+      )
+    end
+  end
+
+  defp run_pipeline(sources, result_directory) do
+    {:ok, pipeline} =
+      %Testing.Pipeline.Options{
+        module: TestPipeline,
+        custom_args: %{
+          sources: sources,
+          output_dir: result_directory
+        }
+      }
+      |> Testing.Pipeline.start_link()
+
+    Testing.Pipeline.play(pipeline)
+    assert_pipeline_playback_changed(pipeline, _, :playing)
+
+    for _source <- sources,
+        do: assert_end_of_stream(pipeline, :sink_bin, {Membrane.Pad, :input, _source})
+
+    Testing.Pipeline.stop_and_terminate(pipeline, blocking?: true)
+    assert_pipeline_playback_changed(pipeline, _, :stopped)
+  end
+
+  defp test_pipeline(sources, reference_directory, test_directory) do
+    hackney_sources =
+      sources
+      |> Enum.map(fn {path, encoding, name} ->
+        {%Membrane.Hackney.Source{location: path, hackney_opts: [follow_redirect: true]},
+         encoding, name}
+      end)
+
+    if @create_fixtures do
+      run_pipeline(hackney_sources, reference_directory)
+    end
+
+    on_exit(fn ->
+      {:ok, test_playlist_content} = File.ls(test_directory)
+
+      for file_name <- test_playlist_content |> Enum.filter(&(!String.starts_with?(&1, "."))),
+          do: File.rm!(test_directory <> file_name)
+    end)
+
+    run_pipeline(hackney_sources, test_directory)
+
+    {:ok, reference_playlist_content} = File.ls(reference_directory)
+
+    for file_name <- reference_playlist_content,
+        do:
+          assert(
+            File.read!(reference_directory <> file_name) ==
+              File.read!(test_directory <> file_name)
+          )
+  end
```

Comment:
WDYT about such refactor? Running the pipeline twice when `@create_fixtures` is true doesn't make sense anyways. With such a code it is clear that we are either testing or preparing fixtures without actual testing. 
```suggestion
   on_exit(fn ->
      {:ok, test_playlist_content} = File.ls(test_directory)

      for file_name <- test_playlist_content |> Enum.filter(&(!String.starts_with?(&1, "."))),
          do: File.rm!(test_directory <> file_name)
    end)

    if @create_fixtures do
      run_pipeline(hackney_sources, reference_directory)
    else
      run_pipeline(hackney_sources, test_directory)

      {:ok, reference_playlist_content} = File.ls(reference_directory)

      for file_name <- reference_playlist_content,
          do:
            assert(
              File.read!(reference_directory <> file_name) ==
                File.read!(test_directory <> file_name)
            )
    end
```

Reply:
Done